include/deal.II/physics/notation.h

```diff
@@ -0,0 +1,1550 @@
+          Assert(component_n < 1, ExcIndexRange(component_n,0,1));
+
+          switch (component_n)
+            {
+            case (0):
+              return std::make_pair(0u,0u);
+            default:
+              break;
+            }
+
+          AssertThrow(false, ExcInternalError());
+          return std::pair<unsigned int, unsigned int> ();
+        }
+
+
+        template<>
+        std::pair<unsigned int, unsigned int>
+        indices_from_component<2> (const unsigned int component_n,
+                                   const bool         symmetric)
+        {
+          if (symmetric == true)
+            {
+              Assert((component_n < SymmetricTensor<2,2>::n_independent_components),
+                     ExcIndexRange(component_n,0,SymmetricTensor<2,2>::n_independent_components));
+            }
+          else
+            {
+              Assert((component_n < Tensor<2,2>::n_independent_components),
+                     ExcIndexRange(component_n,0,Tensor<2,2>::n_independent_components));
+            }
+
+          switch (component_n)
+            {
+            case (0):
+              return std::make_pair(0u,0u);
+            case (1):
+              return std::make_pair(1u,1u);
+            case (2):
+              return std::make_pair(0u,1u);
+            case (3):
+              Assert(symmetric == false,
+                     ExcIndexRange(component_n,0,SymmetricTensor<2,2>::n_independent_components));
+              return std::make_pair(1u,0u);
+            default:
+              break;
+            }
+
+          AssertThrow(false, ExcInternalError());
+          return std::pair<unsigned int, unsigned int> ();
+        }
+
+
+        template<>
+        std::pair<unsigned int, unsigned int>
+        indices_from_component<3> (const unsigned int component_n,
+                                   const bool         symmetric)
+        {
+          if (symmetric == true)
+            {
+              Assert((component_n < SymmetricTensor<2,3>::n_independent_components),
+                     ExcIndexRange(component_n,0,SymmetricTensor<2,3>::n_independent_components));
+            }
+          else
+            {
+              Assert((component_n < Tensor<2,3>::n_independent_components),
+                     ExcIndexRange(component_n,0,Tensor<2,3>::n_independent_components));
+            }
+
+          switch (component_n)
+            {
+            case (0):
+              return std::make_pair(0u,0u);
+            case (1):
+              return std::make_pair(1u,1u);
+            case (2):
+              return std::make_pair(2u,2u);
+            case (3):
+              return std::make_pair(1u,2u);
+            case (4):
+              return std::make_pair(0u,2u);
+            case (5):
+              return std::make_pair(0u,1u);
+            case (6):
+              Assert(symmetric == false,
+                     ExcIndexRange(component_n,0,SymmetricTensor<2,3>::n_independent_components));
```

Comment:
isn't this in essence the same assertion that would already have triggered above?

Reply:
Indeed. I've made your suggested change!